### PR TITLE
Update Lottie to 4.4.1

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/414235014887631/1206751893829139/f
CC: @THISISDINOSAUR 

**Description**:
Update Lottie to version 4.4.1 to align with macOS and comply with [Apple's Privacy manifest for 3rd party SDK](https://developer.apple.com/support/third-party-SDK-requirements/).

I have attached a video of the Fire animation on iPhone X (I don’t have older phones), as there were questions about the performance of the new version of the Library.

https://github.com/duckduckgo/iOS/assets/1089358/bd8b7237-72e4-4249-99b0-9bab20e97770

**Steps to test this PR**:
1. Run the App.
2. Check fire/whirlpool/airstream animation.
3. Check Shield animation when removing site protection.
4. Check Shield animation when enabling site protection.

**Orientation Testing**:

* [x] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [x] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [x] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/PrivacyIconView.swift
Comment (alessandroboron) [inline DuckDuckGo/PrivacyIconView.swift:57]: Lottie warning: 'LRUAnimationCache' is deprecated: Use DefaultAnimationCache instead, which is thread-safe and automatically responds to memory pressure.

**Review by bwaresiak — APPROVED** [review]
LGTM, but maybe we should wait for upcoming release before merging?

**Pull request merged**
